redwood-treehouse-guest/src/commonMain/kotlin/app/cash/redwood/treehouse/treehouseCompose.kt

```diff
@@ -119,24 +126,25 @@ private class RedwoodZiplineTreehouseUi(
   }
 }
 
-private fun OnBackPressedDispatcherService.asNonService(): OnBackPressedDispatcher {
-  return object : OnBackPressedDispatcher {
-    override fun addCallback(onBackPressedCallback: OnBackPressedCallback): Cancellable {
-      return this@asNonService.addCallback(onBackPressedCallback.asService())
-    }
+private fun ZiplineTreehouseUi.Host.asOnBackPressedDispatcher() = object : OnBackPressedDispatcher {
+  override fun addCallback(onBackPressedCallback: OnBackPressedCallback): Cancellable {
+    return this@asOnBackPressedDispatcher.addOnBackPressedCallback(
+      onBackPressedCallback.asService(),
+    )
   }
 }
 
-private fun OnBackPressedCallback.asService(): OnBackPressedCallbackService {
-  return object : OnBackPressedCallbackService {
-    override var isEnabled: Boolean
-      get() = this@asService.isEnabled
-      set(value) {
-        this@asService.isEnabled = value
-      }
+private fun OnBackPressedCallback.asService() = object : OnBackPressedCallbackService {
+  override var isEnabled: Boolean by this@asService::isEnabled
```

Comment:
This is dramatically more expensive than it seems because it requires a `KProperty` to be created and used to read/write to the property. It's more verbose, but this would be much more efficient:
```suggestion
  override var isEnabled: Boolean
    get() = this@asService.isEnabled
    set(value) { this@asService.isEnabled = value }
```